components/Banner/examples/supplemental/MDCBannerViewLayout.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+// These enums is only used for debugging right now
+typedef NS_ENUM(NSInteger, MDCBannerViewLayoutMode) {
+  MDCBannerViewLayoutSingleLineStyle = 0,
+  MDCBannerViewLayoutMultiLineStackedButtonStyle = 1,
+  MDCBannerViewLayoutMultiLineAlignedButtonStyle = 2
+};
+
+@class MDCBannerViewLayout;
+
+@protocol MDCBannerViewLayoutDataSource <NSObject>
+
+- (UILabel *)textLabelForBannerViewLayout:(MDCBannerViewLayout *)bannerViewLayout;
+- (UIImageView *)iconImageViewForBannerViewLayout:(MDCBannerViewLayout *)bannerViewLayout;
+- (NSInteger)numberOfButtonsForBannerViewLayout:(MDCBannerViewLayout *)bannerViewLayout;
+- (UIButton *)bannerViewLayout:(MDCBannerViewLayout *)bannerViewLayout
+                 buttonAtIndex:(NSInteger)index;
+
+@end
+
+@interface MDCBannerViewLayout : NSObject
```

Comment:
Have you considered using out auto-layout?

Reply:
Mark for discussion.